### DISCUSSION TITLE
os.cp的实际效果似乎有点反直觉

### DISCUSSION THREAD

**Discussion by augustheart** [xmake-io/xmake#3037] [Q&A] [answered]
这几天在编译wxwidgets（需要自己搞的原因在于，像这种规模的框架，往往得自己裁剪，别人不见得适合自己用的）。
最后发现os.cp的行为有点违反直觉。os.cp如果复制一个目录到另外一个目录，如果目的地有一个同名的目录，直觉认为会合并这两个目录，但是实际上的表现是将目的地的同名目录删除后替换了。
以wxwidgets为例，我需要将lib/include/i686xxxxx/下面的wx这个目录合并进include中，这个目录下面有setup.h以及msw目录（msw目录下有一个文件）。
而wxwidgets的默认include下面是wx-3.2/wx/...，这其中wx里面也有msw目录。我在之前已经去掉了wx-3.2目录，将wx目录移到include之下，这里是成功的。
之后使用的cp参数是：os.cp("xxxx/lib/include/i686-xxxx/wx","include")(package:getinstalldir()部分略去了）
最后的结果是include/wx/msw目录完全被替换成lib下面的版本，其中原有的文件都被删除了。

**Comment by waruqi** [comment, marked as answer]
> 认为会合并这两个目录

加 `/*` 才会合并

`os.cp("dir/*", "/dir2")`

> 但是实际上的表现是将目的地的同名目录删除后替换了。

不会删，只会 copy 进去

```
ruki:tmp ruki$ mkdir bar
ruki:tmp ruki$ mkdir foo
ruki:tmp ruki$ touch foo/test
ruki:tmp ruki$ xmake l os.cp foo bar
ruki:tmp ruki$ tree bar
bar
└── foo
    └── test

1 directory, 1 file
```

**Comment by augustheart** [comment]
根据提示，将代码改成了
cp "lib/.../wx/*","include/wx/"后测试成功。
老实说，我个人还是挺迷糊的……
